Sources/Interpreter/Interpreter.swift

```diff
@@ -338,14 +338,14 @@ public struct Interpreter {
 
   /// Returns builtin value stored at `a`.
   func builtinValue(at a: Address) -> BuiltinValue {
-    precondition(a.memoryLayout.type.isBuiltin)
-    let allocation = a.memoryAddress.allocation
-    let offset = a.memoryAddress.offset
-    let typeOfValue = a.memoryLayout.type.base as! BuiltinType
-    return memory[allocation].builtinValue(at: offset, ofType: typeOfValue)
+    precondition(a.type.type.isBuiltin)
+    let i = a.startLocation.allocation
+    let o = a.startLocation.offset
+    let t = a.type.type.base as! BuiltinType
+    return memory[i].builtinValue(at: o, ofType: t)
```

Comment:
I know what you've done here comes from my suggestion, but maybe it would be better to call what is currently `a.type` `a.structure` or simply `a.layout` (sometimes resisting all the type information is counterproductive) and expose `a.type.type` as a computed `a.type` property.

```suggestion
    precondition(a.type.isBuiltin)
    let i = a.startLocation.allocation
    let o = a.startLocation.offset
    let t = a.type.base as! BuiltinType
    return memory[i].builtinValue(at: o, ofType: t)
```